RequestManagementSystem/Client/Request.py

```diff
@@ -550,22 +553,34 @@ def optimize( self ):
           removeRepList.append( ( op2.TargetSE, op2 ) )
           del self.__operations__[i]
           del self.__operations__[i]
-          continue
+          print 'Found sequence starting at %d' % i
+          optFound = True
+      if not optFound:
```

Comment:
I am afraid the algorithm doesn't work in case your Request is only made of the 2 Operations ReplicateAndRegister and RemoveReplica.
First loop, i =0, you put the 2 operations in your lists, and remove them from the operation list. Since optFound is True, we don't increment i, and go back to the beginning of the loop. 
Then at the start of the second loop, i is still 0, ben len(self.__operations__) is 0, so you exit the loop, without ever inserting back the operations.
I just tried it on a toy example, and it is indeed the case.

May I recommend you to write unit test for this method ? No mock required, and VERY useful for future evolution ;-) I can even help there if you want